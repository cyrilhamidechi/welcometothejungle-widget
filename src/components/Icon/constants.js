export const NAMES = [
  "arrowLeft",
  "arrowRight",
  "cancel",
  "logo",
  "picture",
  "play",
  "quote"
];
