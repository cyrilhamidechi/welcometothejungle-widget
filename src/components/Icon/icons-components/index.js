import PropTypes from "prop-types";
import React, { PureComponent } from "react";

import ArrowLeft from "./ArrowLeft";
import ArrowRight from "./ArrowRight";
import Close from "./Close";
import Logo from "./Logo";
import Picture from "./Picture";
import Play from "./Play";
import Quote from "./Quote";

class Icon extends PureComponent {
  static propTypes = {
    tag: PropTypes.string.isRequired
  };

  components = {
    ArrowLeft,
    ArrowRight,
    Close,
    Logo,
    Picture,
    Play,
    Quote
  };

  render() {
    const tagName =
      this.props.tag.charAt(0).toUpperCase() + this.props.tag.slice(1);
    const TagName = this.components[tagName];
    const finalProps = { ...this.props };
    delete finalProps.tag;
    return this.props.tag ? <TagName {...finalProps} /> : undefined;
  }
}
export default Icon;
