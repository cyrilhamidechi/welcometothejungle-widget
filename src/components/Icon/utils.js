import { colors } from "./../../config";

export const color = color => {
  switch (color) {
    case "white":
      return colors.white;
    default:
      return `${color}`;
  }
};
