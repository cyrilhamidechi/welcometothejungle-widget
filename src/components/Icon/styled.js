import styled, { css } from "styled-components";

import { color } from "./utils";

export const Wrapper = styled.span`
  display: inline-block;
  line-height: 1;
  svg {
    width: 100%;
    height: 100%;
  }

  path {
    transition: fill 300ms;
  }

  ${props =>
    props.size &&
    css`
      height: ${props.size}px;

      svg {
        width: ${props.size}px;
        height: ${props.size}px;
      }
    `};

  ${props =>
    props.color &&
    css`
      svg path {
        fill: ${color(props.color)}
    `};
`;
