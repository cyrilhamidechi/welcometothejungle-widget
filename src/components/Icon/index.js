// to generate icons components
// npx @svgr/cli --icon -d src/components/Icon/icons-components src/components/Icon/icons-files/

import PropTypes from "prop-types";
import React, { PureComponent } from "react";

import Icons from "./icons-components/";

import { NAMES } from "./constants";

import * as S from "./styled";

class Icon extends PureComponent {
  static propTypes = {
    color: PropTypes.string,
    name: PropTypes.oneOf(NAMES).isRequired,
    size: PropTypes.string
  };

  getIcon(name, title) {
    const iconProps = {
      // accessibility
      title: name,
      role: "img",
      "aria-labelledby": "title"
    };

    const icon = NAMES.find(index => name === index);
    return <Icons tag={icon} {...iconProps} />;
  }

  render() {
    const { color, name, size, title } = this.props;
    return (
      <S.Wrapper color={color} size={size}>
        {this.getIcon(name, title)}
      </S.Wrapper>
    );
  }
}

export default Icon;
