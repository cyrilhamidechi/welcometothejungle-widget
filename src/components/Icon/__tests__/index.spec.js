import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import Icon from "../";

it("should render the component correctly with only required props", () => {
  const tree = renderer.create(<Icon name="arrowLeft" />).toJSON();

  expect(tree).toMatchSnapshot();
});

it("should render the component correctly with size props", () => {
  const tree = renderer.create(<Icon name="arrowLeft" size="30" />).toJSON();

  expect(tree).toMatchSnapshot();
});

it("should render the component correctly with color props", () => {
  const tree = renderer
    .create(<Icon name="arrowLeft" color="white" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
