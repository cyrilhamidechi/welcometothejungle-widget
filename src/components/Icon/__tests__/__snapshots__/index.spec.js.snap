// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component correctly with color props 1`] = `
.c0 {
  display: inline-block;
  line-height: 1;
}

.c0 svg {
  width: 100%;
  height: 100%;
}

.c0 path {
  -webkit-transition: fill 300ms;
  transition: fill 300ms;
}

.c0 svg path {
  fill: #fff;
}

<span
  className="c0"
  color="white"
>
  <svg
    aria-labelledby="title"
    height="1em"
    role="img"
    title="arrowLeft"
    viewBox="0 0 24 24"
    width="1em"
  >
    <path
      d="M19 11H7.83l4.88-4.88c.39-.39.39-1.03 0-1.42a.996.996 0 0 0-1.41 0l-6.59 6.59a.996.996 0 0 0 0 1.41l6.59 6.59a.996.996 0 1 0 1.41-1.41L7.83 13H19c.55 0 1-.45 1-1s-.45-1-1-1z"
    />
  </svg>
</span>
`;

exports[`should render the component correctly with only required props 1`] = `
.c0 {
  display: inline-block;
  line-height: 1;
}

.c0 svg {
  width: 100%;
  height: 100%;
}

.c0 path {
  -webkit-transition: fill 300ms;
  transition: fill 300ms;
}

<span
  className="c0"
>
  <svg
    aria-labelledby="title"
    height="1em"
    role="img"
    title="arrowLeft"
    viewBox="0 0 24 24"
    width="1em"
  >
    <path
      d="M19 11H7.83l4.88-4.88c.39-.39.39-1.03 0-1.42a.996.996 0 0 0-1.41 0l-6.59 6.59a.996.996 0 0 0 0 1.41l6.59 6.59a.996.996 0 1 0 1.41-1.41L7.83 13H19c.55 0 1-.45 1-1s-.45-1-1-1z"
    />
  </svg>
</span>
`;

exports[`should render the component correctly with size props 1`] = `
.c0 {
  display: inline-block;
  line-height: 1;
  height: 30px;
}

.c0 svg {
  width: 100%;
  height: 100%;
}

.c0 path {
  -webkit-transition: fill 300ms;
  transition: fill 300ms;
}

.c0 svg {
  width: 30px;
  height: 30px;
}

<span
  className="c0"
  size="30"
>
  <svg
    aria-labelledby="title"
    height="1em"
    role="img"
    title="arrowLeft"
    viewBox="0 0 24 24"
    width="1em"
  >
    <path
      d="M19 11H7.83l4.88-4.88c.39-.39.39-1.03 0-1.42a.996.996 0 0 0-1.41 0l-6.59 6.59a.996.996 0 0 0 0 1.41l6.59 6.59a.996.996 0 1 0 1.41-1.41L7.83 13H19c.55 0 1-.45 1-1s-.45-1-1-1z"
    />
  </svg>
</span>
`;
